### PR TITLE
implemented SASX and SSAX correctly.

### PR DESCRIPTION
**Pull request by FernandoS27**

### CODE REVIEW

File: src/backend_x64/emit_x64.cpp [hunk 1]
```diff
@@ -1682,6 +1682,65 @@ void EmitX64::EmitPackedSubS16(IR::Block& block, IR::Inst* inst) {
...
+    else
+        code->ror(reg_b, 16);
+    if (ge_inst)
+        code->movd(xmm_b, reg_b);
+    code->movd(xmm_a, reg_a);
+    if (ge_inst) {
+        Xbyak::Xmm xmm_ge = reg_alloc.ScratchXmm();
+        Xbyak::Xmm xmm_tmp = reg_alloc.ScratchXmm();
+
+        // TODO: find a simpler way to calculate ge flag
```
Comment (MerryMage) [inline src/backend_x64/emit_x64.cpp]: You might as well do the addition and subtraction separately at this point.

Comment (FernandoS27) [inline src/backend_x64/emit_x64.cpp]: hmn, I prefered keeping the main implementation simple as it matches the interpreter in all known cases, while the ge instruction giving overhead. After all, most ge_inst are eliminated by the optimizer.

Comment (MerryMage) [inline src/backend_x64/emit_x64.cpp]: That's far; I guess an improved version can be implemented later.

**Comment by MerryMage** [comment]
Subtract is spelled `Subtract`, not `Substract`.

**Comment by lioncash** [comment]
Also, the second instruction is SSAX, not SSASX.

File: src/frontend/ir/opcodes.inc [hunk 2]
```diff
@@ -89,6 +89,7 @@ OPCODE(PackedAddU16,            T::U32,         T::U32,         T::U32
 OPCODE(PackedAddS16,            T::U32,         T::U32,         T::U32                          )
 OPCODE(PackedSubU16,            T::U32,         T::U32,         T::U32                          )
 OPCODE(PackedSubS16,            T::U32,         T::U32,         T::U32                          )
+OPCODE(PackedSubtractAddS16,   T::U32,         T::U32,         T::U32,         T::U1           )
```
Comment (lioncash) [inline src/frontend/ir/opcodes.inc]: Spacing is off here.

**Comment by Dragios** [comment]
What is the status of this PR?

**Comment by MerryMage** [comment]
Closed by 05e97058c32fab0e0ebf88e7fff2a4b77890e20

**Pull request closed without merging**